### PR TITLE
add c code for chapter_computational_complexity

### PR DESCRIPTION
**Pull request by sjinzh**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/c/chapter_computational_complexity/time_complexity.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
+// File: time_complexity.c
+// Created Time: 2023-01-03
+// Author: sjinzh (sjinzh@gmail.com)
+
+#include "stdio.h"
```
Comment (krahets) [inline codes/c/chapter_computational_complexity/time_complexity.c:5]: Replace them with `#include "../include/include.h"`

File: codes/c/chapter_computational_complexity/time_complexity.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
+// File: time_complexity.c
+// Created Time: 2023-01-03
+// Author: sjinzh (sjinzh@gmail.com)
```
Comment (krahets) [inline codes/c/chapter_computational_complexity/time_complexity.c:3]: Please add the code to the docs (markdown file).

File: codes/c/chapter_computational_complexity/time_complexity.c [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+#include "stdio.h"
+#include "stdlib.h"
+
+// 常数阶
```
Comment (krahets) [inline codes/c/chapter_computational_complexity/time_complexity.c:8]: We use `/* 常数阶 */` to comment on a function and a class.

File: codes/c/chapter_computational_complexity/time_complexity.c (hunk 1 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/time_complexity.c]: done.

File: codes/c/chapter_computational_complexity/time_complexity.c (hunk 2 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/time_complexity.c]: done.

File: codes/c/chapter_computational_complexity/time_complexity.c (hunk 1 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/time_complexity.c]: > Replace them with `#include "../include/include.h"`

done.

File: codes/c/chapter_computational_complexity/time_complexity.c (hunk 2 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/time_complexity.c]: > Please add the code to the docs (markdown file).

done.

File: codes/c/chapter_computational_complexity/time_complexity.c (hunk 3 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/time_complexity.c]: > We use `/* 常数阶 */` to comment on a function and a class.

done.

File: codes/c/chapter_computational_complexity/worst_best_time_complexity.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+{
+    // 分配堆区内存（创建一维可变长数组：数组中元素数量为n，元素类型为int）
+    int *nums = (int *)malloc(n * sizeof(int));
+    // 生成数组 nums = { 1, 2, 3, ..., n }
+    for (int i = 0; i < n; i++)
+    {
+        nums[i] = i + 1;
+    }
+    // 随机打乱数组元素 
+	for (int i = n - 1; i > 0; i--)
```
Comment (krahets) [inline codes/c/chapter_computational_complexity/worst_best_time_complexity.c:20]: Incorrect indentation

File: docs/chapter_computational_complexity/time_complexity.md [hunk 5]
```diff
@@ -2061,7 +2225,62 @@ $$
...
+        srand((unsigned int)time(NULL));
+        for (int i = 0; i < 10; i++)
+        {
+            int n = 100;
+            int *nums = randomNumbers(n);
+            int index = findOne(nums, n);
+            printf("\n数组 [ 1, 2, ..., n ] 被打乱后 = ");
+            for (int j = 0; j < n; j++)
+            {
+                printf("%d%s", nums[j], (j == n-1)? "" : "," );
```
Comment (krahets) [inline docs/chapter_computational_complexity/time_complexity.md:2272]: Could we add a `printArray()` function in the `include/PrintUtil.c`?

File: docs/chapter_computational_complexity/time_complexity.md [hunk 6]
```diff
@@ -2061,7 +2225,62 @@ $$
...
+            }
+            printf("\n数字 1 的索引为 %d\n", index);
+            // 释放堆区内存
+            if(nums != NULL){
```
Comment (krahets) [inline docs/chapter_computational_complexity/time_complexity.md:2276]: `if (nums != NULL) {`

Please check the formats

File: docs/chapter_computational_complexity/time_complexity.md [hunk 7]
```diff
@@ -1012,7 +1090,18 @@ $$
 === "C"
 
     ```c title="time_complexity.c"
-
+    /* 平方阶 */
+    int quadratic(int n) 
+    {
+        int count = 0;
+        // 循环次数与数组长度成平方关系
+        for(int i = 0; i < n; i++){
```
Comment (krahets) [inline docs/chapter_computational_complexity/time_complexity.md:1098]: `for (int i = 0; i < n; i++) {`

File: docs/chapter_computational_complexity/time_complexity.md [hunk 8]
```diff
@@ -1163,7 +1252,25 @@ $$
...
-
+    /* 平方阶（冒泡排序） */
+    int bubbleSort(int *nums, int n)
+    {
+        int count = 0;  // 计数器 
+        // 外循环：待排序元素数量为 n-1, n-2, ..., 1
+        for(int i = n - 1; i > 0; i--){
+            // 内循环：冒泡操作
+            for (int j = 0; j < i; j++)
+            {
```
Comment (krahets) [inline docs/chapter_computational_complexity/time_complexity.md:1263]: Please unify the location of `{`.

File: docs/chapter_computational_complexity/time_complexity.md [hunk 9]
```diff
@@ -1163,7 +1252,25 @@ $$
...
+                // 交换 nums[j] 与 nums[j + 1]
+                int tmp = nums[j];
+                nums[j] = nums[j + 1];
+                nums[j + 1] = tmp;
+                count += 3;  // 元素交换包含 3 个单元操作
+            }
+
```
Comment (krahets) [inline docs/chapter_computational_complexity/time_complexity.md:1270]: Remove the empty line.

File: codes/c/chapter_computational_complexity/worst_best_time_complexity.c [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        int *nums = randomNumbers(n);
+        int index = findOne(nums, n);
+        printf("\n数组 [ 1, 2, ..., n ] 被打乱后 = ");
+        for (int j = 0; j < n; j++)
+        {
+            printf("%d%s", nums[j], (j == n-1)? "" : "," );
+        }
+        printf("\n数字 1 的索引为 %d\n", index);
+        // 释放堆区内存
+        if(nums != NULL){
```
Comment (krahets) [inline codes/c/chapter_computational_complexity/worst_best_time_complexity.c:57]: `if (nums != NULL) {` Same for others.

**Review by krahets — COMMENTED** [review]
Please address the comments.

File: codes/c/chapter_computational_complexity/worst_best_time_complexity.c (hunk 4 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/worst_best_time_complexity.c]: Please see PR #208.

File: docs/chapter_computational_complexity/time_complexity.md (hunk 5 above)
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR https://github.com/krahets/hello-algo/pull/208.

File: docs/chapter_computational_complexity/time_complexity.md (hunk 6 above)
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR #208.

File: docs/chapter_computational_complexity/time_complexity.md (hunk 7 above)
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR #208.

File: docs/chapter_computational_complexity/time_complexity.md (hunk 9 above)
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR #208.

File: codes/c/chapter_computational_complexity/worst_best_time_complexity.c (hunk 10 above)
Comment (sjinzh) [reply codes/c/chapter_computational_complexity/worst_best_time_complexity.c]: Please see PR #208.

File: docs/chapter_computational_complexity/time_complexity.md (hunk 8 above)
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR #208.

File: docs/chapter_computational_complexity/time_complexity.md [hunk 11]
```diff
@@ -1163,7 +1252,25 @@ $$
 === "C"
 
     ```c title="time_complexity.c"
-
+    /* 平方阶（冒泡排序） */
+    int bubbleSort(int *nums, int n)
+    {
+        int count = 0;  // 计数器 
+        // 外循环：待排序元素数量为 n-1, n-2, ..., 1
+        for(int i = n - 1; i > 0; i--){
```
[the remark this answers is not in the document]
Comment (sjinzh) [reply docs/chapter_computational_complexity/time_complexity.md]: Please see PR #208.

**Pull request closed without merging**